### ISSUE TITLE
Refactor the OAI code, Step 2

### ISSUE THREAD

**Issue by landreev** [IQSS/dataverse#8842]
This is a followup issue for #8372 (PR #8753). 
Once that is merged we can proceed cleaning up and refactoring the remaining old OAI code. 
Some of that code predates Dataverse and was copy-and-pasted from the implementation in DVN. 
In the process it created in some extra layers of code that duplicate each other, and should probably be significantly simplified. (That is on top of a significant amount of old code that we used to maintain on the Dataverse side, but got rid of in Step. 1 when switching to the new gdcc/xoai library). 
Major pieces: 
- Server: the object implementing XOAI "items" (DataverseXoaiItem) acts as a wrapper around the entity class storing the record in the database (OAIRecord). The XOAI items should instead implement both. There's a similar situation with the Set objects. 
- Client: OaiHandler may not be necessary/may be duplicating the functionality provided by the XOAI library; should be possible to further streamline the implementation.

Aside from that, the overall code needs to be cleaned up and modernized.

**Comment by poikilotherm** [comment]
- Add a nice XSLT to the response. Needs an endpoint to serve the XSLT from a `String` at `DataProvider.getOaiXslt()` and a processing instruction with the `XmlWriter` in `OAIServlet`.

**Comment by mreekie** [comment]
sizing:
- This needs to be split out into smaller issues.
- we will take a swipe at this during the discussion of 1.4.1
- skipping sizing for now.

**Comment by landreev** [comment]
I will make an attempt to create the child issue for the identifiable sub-tasks involved.

**Comment by mreekie** [comment]
Sizing: 
- I recommend that we split off a spike from this to capture the work Leonid has proposed

**Comment by mreekie** [comment]
Sizing:
- The suggested spike is part of: https://github.com/IQSS/dataverse/issues/8574
- No sizing today.

**Comment by mreekie** [comment]
Sizing:
- Leonid will step through issues before monday and create a plan/break off issues

**Comment by cmbz** [comment]
@landreev I'm following up on @mreekie's comment from 19 April. 
- Can you share links to the planned issues he mentions?

**Comment by cmbz** [comment]
@landreev Following up on @mreekie's comment from [19 April](https://github.com/IQSS/dataverse/issues/8842#issuecomment-1514933403): Could you share links to the issues he mentions related to this issue: https://github.com/IQSS/dataverse/issues/8842?

**Comment by cmbz** [comment]
2023/09/25: Added to 6.1 milestone as per conversation during prioritization meeting.

**Comment by cmbz** [comment]
2024/03/12
Unfortunately, this issue won't be addressed in 6.2 but I will add it as a 6.3 milestone instead.

**Comment by cmbz** [comment]
2024/11/21
- Assigning to @landreev and placing On Hold for now. 
- @landreev please decide whether you want this to go back into the backlog or close it and revisit the topic at a future date.
